resources/web/wwi/server.js

```diff
@@ -0,0 +1,96 @@
+/* global webots, Stream */
+'use strict';
+
+function Server(url, view, onready) {
+  this.view = view;
+  this.onready = onready;
+
+  // url has the following form: "ws(s)://cyberbotics2.cyberbotics.com:80/simple/worlds/simple.wbt"
+  var n = url.indexOf('/', 6);
+  var m = url.lastIndexOf('/');
+  this.url = 'http' + url.substring(2, n); // e.g., "http(s)://cyberbotics2.cyberbotics.com:80"
+  this.project = url.substring(n + 1, m - 7); // e.g., "simple"
+  this.worldFile = url.substring(m + 1); // e.g., "simple.wbt"
+  this.controllers = [];
+};
+
+Server.prototype = {
+  constructor: Server,
+
+  connect: function() {
+    var that = this;
+    var xhr = new XMLHttpRequest();
+    xhr.open('GET', this.url + '/session', true);
+    $('#webotsProgressMessage').html('Connecting to session server...');
+    xhr.onreadystatechange = function(e) {
+      if (xhr.readyState !== 4)
+        return;
+      if (xhr.status !== 200)
+        return;
+      var data = xhr.responseText;
+      if (data.startsWith('Error:')) {
+        $('#webotsProgress').hide();
+        var errorMessage = data.substring(6).trim();
+        errorMessage = errorMessage.charAt(0).toUpperCase() + errorMessage.substring(1);
+        webots.alert('Session server error', errorMessage);
+        return;
+      }
+      that.socket = new WebSocket(data + '/client');
+      that.socket.onopen = function(event) { that.onOpen(event); };
+      that.socket.onmessage = function(event) { that.onMessage(event); };
+      that.socket.onclose = function(event) {
+        that.view.console.info('Disconnected to the Webots server.');
+      };
+      that.socket.onerror = function(event) {
+        that.view.console.error('Cannot connect to the simulation server');
+      };
+    };
+    xhr.send();
+  },
+
+  onOpen: function(event) {
+    var host = location.protocol + '//' + location.host.replace(/^www./, ''); // remove 'www' prefix
+    if (typeof webots.User1Id === 'undefined')
+      webots.User1Id = '';
+    if (typeof webots.User1Name === 'undefined')
+      webots.User1Name = '';
+    if (typeof webots.User1Authentication === 'undefined')
+      webots.User1Authentication = '';
+    if (typeof webots.User2Id === 'undefined')
+      webots.User2Id = '';
+    if (typeof webots.User2Name === 'undefined')
+      webots.User2Name = '';
+    if (typeof webots.CustomData === 'undefined')
```

Comment:
There are currently 2 syntaxes to check against 'undefined':

```
typeof data === 'undefined'
```

or 

```
data === undefined
```

The first syntax is often preferred. My concern is just to be coherent. https://stackoverflow.com/questions/3390396/how-to-check-for-undefined-in-javascript